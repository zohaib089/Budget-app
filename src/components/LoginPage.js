import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Budget App</h1>
      <p>Manage your Budget properly and keep track of it</p>
      <button className="button" onClick={startLogin}>
        Login with Google
      </button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
