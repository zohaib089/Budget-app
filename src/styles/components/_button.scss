.button {
  background: $blue_violet;
  border: none;
  color: white;
  padding: $s-size;
  font-weight: 300;
  font-size: $font-size-large;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
}
.button--link {
  background: none;
}
