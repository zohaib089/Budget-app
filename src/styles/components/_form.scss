.form {
  display: flex;
  flex-direction: column;
  > * {
    margin-bottom: $m-size;
  }
}

.form__error {
  margin: 0 0 $m-size 0;
  font-style: italic;
}

.button__secondary {
  background: grey;
}
