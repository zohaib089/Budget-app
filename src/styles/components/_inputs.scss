.text-input {
  border: 1px solid wheat;
  height: 50px;
  padding: $s-size;
  font-size: $font-size-large;
  font-weight: 300;
}
.select {
  border: 1px solid wheat;
  height: 50px;
}

.textarea {
  @extend .text-input;
  height: 10rem;
}
