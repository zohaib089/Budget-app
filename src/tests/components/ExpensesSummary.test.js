import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render expense summary with 1 expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={275} />
  );
  expect(wrapper).toMatchSnapshot();
});
test('should render expense summary with mulitple expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={23} expensesTotal={27512345678} />
  );
  expect(wrapper).toMatchSnapshot();
});
