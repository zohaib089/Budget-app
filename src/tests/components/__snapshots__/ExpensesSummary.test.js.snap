// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary with 1 expense 1`] = `
<div>
  <h1>
    Showing 
    1
     
    expense
     total is 
    $2.75
  </h1>
</div>
`;

exports[`should render expense summary with mulitple expense 1`] = `
<div>
  <h1>
    Showing 
    23
     
    expenses
     total is 
    $275,123,456.78
  </h1>
</div>
`;
